plugins/Lights/Lights.cpp

```diff
@@ -110,10 +126,23 @@ void Lights::setOffMillisec(int offMs)
 
 bool Lights::init()
 {
-    if (m_lightDevice) {
+    if (m_usesHfd || m_lightDevice) {
+        return true;
+    }
+
+    //Try first to detect if hfd should be used
+    QProcess getpropProcess;
+    qWarning() << "Testing for availability of hfd interface";
+    getpropProcess.start("/usr/bin/getprop", QStringList() << "ro.build.version.sdk");
+    getpropProcess.waitForFinished();
+    int processResult = getpropProcess.readAllStandardOutput().toInt();
```

Comment:
We currently has to link against libhybris anyway. Use `property_get()`.

You probably has to link against `libandroid-properties`. See [this file](https://github.com/ubports/unity8/blob/9ec9925db70d3fb34759cf81ce1272e84176a412/src/CMakeLists.txt).

Reply:
Actually, I guess if we try to initialize the legacy HAL first, and then initialize Hfd interface if failed, we might not have to have this property. Unless there exists a device whose legacy HAL is broken...